app.js

```diff
@@ -1,27 +1,39 @@
 const express = require('express');
 const app = express();
-const {getTopics, getAPI} = require('./controllers/controllers.js')
+const {getTopics, getAPI, getArticlesID} = require('./controllers/controllers.js')
 
 app.get('/api/topics', getTopics);
 app.get('/api', getAPI);
+app.get('/api/articles/:article_id', getArticlesID);
 
 
-// Middleware to catch errors
+// Middleware to catch 404 errors (route not found)
 app.use((req, res, next) => {
-    const err = new Error('Not found');
+    const err = new Error('Invalid input');
     err.status = 404;
+    err.msg = 'Invalid input';
     next(err);
 });
 
-//Error Handling middleware
+// Middleware to handle specific Postgres error codes
 app.use((err, req, res, next) => {
     console.log(err);
-
-    if (err.code === '23502') {
+    if (err.code === '22P02' || err.code === '23502') {
         res.status(400).send({ msg: 'Bad request' });
-    } else if (err.status === 404) {
-        res.status(404).send({ msg: 'Invalid input' });
+    } else {
+        next(err);
     }
 });
 
+// General error handling middleware for known errors with status and message
+app.use((err, req, res, next) => {
+    if (err.status && err.msg) {
+        res.status(err.status).send({ msg: err.msg });
+    } else {
+        next(err);
+    }
+});
```

Comment:
To help streamline your app.js the error handlers can also be extracted into their own file, like your controllers currently are